### ISSUE TITLE
CriticalConfigurationError secure cookie on plain http after upgrade to 1.15.4

### ISSUE THREAD

**Issue by Don-Raf** [simplesamlphp/simplesamlphp#808]
After upgrading to 1.15.4 I started getting "Uncaught SimpleSAML\Error\CriticalConfigurationError: The configuration is invalid: Setting secure cookie on plain HTTP is not allowed. /lib/SimpleSAML/Session.php:296"

In the config.php the following was set and was working fine behind a reverse nginx proxy which terminates the TLS connection:

'baseurlpath' => 'https://FQDN/simplesaml/',
'session.cookie.secure' => true,

After the upgrade I had to change 'session.cookie.secure' to false to prevent the error.

Something that changed between 1.13.2 and 1.15.4 that I should be aware of?
I compared the config.php files and couldn't find anything obvious.

**Comment by jaimeperez** [comment]
Can you provide a full stack trace of the error? Do you get it when you call SimpleSAMLphp for authentication from your application?

If so, you need to set the following configuration option:

```php
    'application' => array(
        'baseURL' => 'https://FQDN'
    ),
```

**Comment by Don-Raf** [comment]
Sorry for the late reply. 

Set up a test installation and verified the 'application' setting worked.
Didn't find that option since the old 1.13.2 installation config.php didn't have it. Copied in from new config and using secure cookie worked again. 

Thanks for the help!

**Comment by modxcustomize** [comment]
> session.cookie.secure

Hi, sorry but what is the FQDN standing for? Is it the URL of the app like http://appdomain.tld ? What exactly it should stand for?

**Comment by thijskh** [comment]
Fully Qualified Domain Name, indeed as you say.

**Comment by modxcustomize** [comment]
> Fully Qualified Domain Name, indeed as you say.

Thanks. So, it is set as 
`'application' => array(
        //'baseURL' => 'https://FQDN'
        'baseURL' => 'http://sso.sergey.....com'
    ),`
did it needs any slash or else, please? The problem I am trying to resolve is that the Admin login keeps redirecting and finally dies with browser error like ERR TOO MANY REDIRECTS

**Comment by modxcustomize** [comment]
> The admin login is something from SimpleSAMLphp itself. The config you reference is in your own aplication, so unrelated.
> 
> I would investigate with the browser debug bar what exactly happens on authentication, where it redirects to and whether there's anything like cookies getting lost or other things you would not expect in the login flow. This requires some debugging from your side, not something we can remotely help with just by describing the symptom.

Thanks a lot. It seems working with the Firefox so I will look a bit more for the Chrome.

**Comment by thijskh** [comment]
Maybe you have a SameSite cookie issue. See the SameSite section in [ Debugging "State Information Lost" errors ](https://simplesamlphp.org/docs/stable/simplesamlphp-nostate.html)

**Comment by modxcustomize** [comment]
> Maybe you have a SameSite cookie issue. See the SameSite section in [ Debugging "State Information Lost" errors ](https://simplesamlphp.org/docs/stable/simplesamlphp-nostate.html)
OK, thanks a lot for the idea. BTW, do you know if this software will be updated maintained and good functioning in php8?

**Comment by thijskh** [comment]
It is maintained and works with PHP 8

**Comment by tvdijen** [comment]
Note that some issues with PHP8+ have been solved recently in 1.19.7, so it's important to use the latest version.
